core/src/main/java/org/apache/carbondata/core/datastore/filesystem/AbstractDFSCarbonFile.java

```diff
@@ -273,71 +263,52 @@ public boolean isFileModified(long fileTimeStamp, long endOffset) {
   }
 
   @Override
-  public DataOutputStream getDataOutputStream(String path, FileFactory.FileType fileType,
-      int bufferSize, boolean append) throws IOException {
-    Path pt = new Path(path);
-    FileSystem fileSystem = pt.getFileSystem(FileFactory.getConfiguration());
+  public DataOutputStream getDataOutputStream(int bufferSize, boolean append)
+      throws IOException {
     FSDataOutputStream stream = null;
     if (append) {
       // append to a file only if file already exists else file not found
       // exception will be thrown by hdfs
-      if (CarbonUtil.isFileExists(path)) {
-        stream = fileSystem.append(pt, bufferSize);
+      if (CarbonUtil.isFileExists(path.toString())) {
```

Comment:
Why not call `isFileExists` directly in `FileFactory`?

Reply:
There are several methods in CarbonUtil which are handling the exceptions or are used to pass common values to FileFactory...so i think its better to leave them as it is and they can be refactored some time later.